### PR TITLE
Fix quitting the project

### PR DESCRIPTION
**Pull request by matthijskooijman**
This makes quitting the project work again, and fixes the digest message on reincarnating a user.

### CODE REVIEW

File: system/datatypes/settings.php [hunk 1]
```diff
@@ -149,14 +149,25 @@ function editQuitMessage() {
...
+				hypha_retireUser($user, /* by_admin */ false);
+				$hyphaXml->saveAndUnlock();
+
+				$subject = __('user-has-left-project', ['name' => $user->getAttribute('fullname'), 'project' => hypha_getTitle()]);
+				$body = nl2br(htmlspecialchars($message));
+				$error = sendMail(getUserEmailList(), $subject, $body);
+				if ($error)
+					notify('error', $error);
+
+				// Leave wil have worked, even when the e-mail failed
```
Comment (laurensmartina) [inline system/datatypes/settings.php]: typo; will

Comment (laurensmartina) [reply system/datatypes/settings.php]: type; email

File: system/core/base.php [hunk 2]
```diff
@@ -459,6 +459,24 @@ function hypha_setUser($user, $username, $password, $fullname, $email, $language
 		}
 	}
 
+	/*
+		Function: hypha_retireUser
+		Retires the given user and adds a note to the digest about this.
+	*/
+	function hypha_retireUser($user, $by_admin) {
```
Comment (laurensmartina) [inline system/core/base.php]: Passing the user_id in stead of the user object is a bit more clear and less chances on passing the wrong user object, IMO.

File: system/datatypes/settings.php (hunk 1 above)
Comment (matthijskooijman) [reply system/datatypes/settings.php]: Fixed.

File: system/core/base.php (hunk 2 above)
Comment (matthijskooijman) [reply system/core/base.php]: Makes sense. Does mean the user must be looked up twice in one of the instances, but that's not really a big deal. Fixed!

**Comment by matthijskooijman** [comment]
Fixed your comments and rebased on top of master, merging now.

**Pull request merged**